### PR TITLE
Standardize default fab perms

### PR DESCRIPTION
**Pull request by jhtimmins**
This PR applies the Stable API permissions model to the existing FAB permission views.

In order to merge, the new `/users` and `/roles` endpoints were updated to use the FAB permissions model.

## The following mapping changes are included
`PermissionModelView.can_list` => `Permissions.can_read`
`PermissionViewModelView.can_list` => `Permission Views.can_read`
`ResetMyPasswordView.can_this_form_get` => `My Password.can_read`
`ResetMyPasswordView.can_this_form_post` => `My Password.can_edit`
`ResetPasswordView.can_this_form_get` => `Passwords.can_read`
`ResetPasswordView.can_this_form_post` => `Passwords.can_edit`
`RoleModelView.can_delete` => `Roles.can_delete`
`RoleModelView.can_download` => `Roles.can_read`
`RoleModelView.can_show` => `Roles.can_read`
`RoleModelView.can_list` => `Roles.can_read`
`RoleModelView.can_edit` => `Roles.can_edit`
`RoleModelView.can_add` => `Roles.can_create`
`RoleModelView.can_copyrole` => `Roles.can_create`
`ViewMenuModelView.can_list` => `View Menus.can_read`
`UserDBModelView.can_add` => `View Menus.can_create`
`UserDBModelView.can_userinfo` => `My Profile.can_read`
`UserDBModelView.can_download` => `View Menus.can_read`
`UserDBModelView.can_show` => `View Menus.can_read`
`UserDBModelView.can_list` => `View Menus.can_read`
`UserDBModelView.can_edit` => `View Menus.can_edit`
`UserDBModelView.resetmypassword` => `My Password.can_read`
`UserDBModelView.resetpasswords` => `Passwords.can_read`
`UserDBModelView.userinfoedit` => `My Profile.can_edit`
`UserDBModelView.can_delete` => `View Menus.can_delete`
`UserInfoEditView.can_this_form_get` => `My Profile.can_read`
`UserInfoEditView.can_this_form_post` => `My Profile.can_edit`
`UserStatsChartView.can_chart` => `User Stats Chart.can_read`
`UserLDAPModelView.can_userinfo` => `My Profile.can_read`
`UserOAuthModelView.can_userinfo` => `My Profile.can_read`
`UserOIDModelView.can_userinfo` => `My Profile.can_read`
`UserRemoteUserModelView.can_userinfo` => `My Profile.can_read`
`DagRunModelView.can_clear` => `Dags.can_read`, `TaskInstances.can_delete`

### CODE REVIEW

File: airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+branch_labels = None
+depends_on = None
+
+
+mapping = {
+    ("PermissionModelView", "can_list"): [
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_PERMISSION),
+    ],
+    ("PermissionViewModelView", "can_list"): [
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_PERMISSION_VIEW),
```
Comment (ephraimbuddy) [inline airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py:43]: Wondering if there's a mistake here on using RESOURCE_PERMISSION_VIEW?

Comment (ephraimbuddy) [reply airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py:43]: Ah...Got it

**Comment by kaxil** [comment]
Rebased to latest master to fix the failing errors around providers

File: airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    ("DagRunModelView", "can_clear"): [
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_DELETE, permissions.RESOURCE_TASK_INSTANCE),
+    ],
+}
+
+
+def remap_permissions():
+    """Apply Map Airflow view permissions."""
+    appbuilder = create_app(config={'FAB_UPDATE_PERMS': False}).appbuilder
```
Comment (ashb) [inline airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py:141]: Should we also set `conf.getboolean('webserver', 'UPDATE_FAB_PERMS'):` somehow?

**Comment by ashb** [comment]
FYI: This PR and https://github.com/apache/airflow/pull/14840 "fight"

**Comment by jhtimmins** [comment]
@ashb This updates the permission model, so #14840 will need to integrate these changes.
Can you and @kaxil take a look?

File: airflow/www/views.py [hunk 3]
```diff
@@ -3806,3 +3822,204 @@ def autocomplete(self, session=None):
...
+
+
+class CustomUserDBModelView(UserDBModelView):
+    """Customize permission names for FAB's builtin UserDBModelView."""
+
+    _class_permission_name = permissions.RESOURCE_USER
+
+    class_permission_name_mapping = {
+        'resetmypassword': permissions.RESOURCE_MY_PASSWORD,
+        'resetpasswords': permissions.RESOURCE_PASSWORD,
```
Comment (kaxil) [inline airflow/www/views.py:3945]: What's the difference between this two? Just want to make sure we don't accidentally give permissions to a user to change password for other users :)

File: airflow/www/views.py [hunk 4]
```diff
@@ -3806,3 +3822,204 @@ def autocomplete(self, session=None):
...
+    method_permission_name = {
+        'chart': 'read',
+    }
+    base_permissions = [permissions.ACTION_CAN_READ]
+
+
+class CustomUserLDAPModelView(UserLDAPModelView):
+    """Customize permission names for FAB's builtin UserLDAPModelView."""
+
+    class_permission_name = permissions.RESOURCE_MY_PROFILE
```
Comment (kaxil) [inline airflow/www/views.py:4014]: Is there a specific reason we call `"/userstatschartview"` as `RESOURCE_USER_STATS_CHART` but for `userinfo` one we call it `RESOURCE_MY_PROFILE`

i.e. `RESOURCE_MY` vs `RESOURCE_USER`. Should we standardize it to say `RESOURCE_USER_STATS_CHART`  and the other one as `RESOURCE_USER_PROFILE`?

**Review by kaxil — COMMENTED** [review]
Minor comments otherwise LGTM

File: airflow/www/views.py (hunk 3 above)
Comment (jhtimmins) [reply airflow/www/views.py:3945]: `resetmypassword` just lets a user update their own password.

`resetpasswords` lets them update any user's password, and should be limited to Admin-level users.

File: airflow/www/views.py (hunk 4 above)
Comment (jhtimmins) [reply airflow/www/views.py:4014]: `userstatschartview` applies to all users, whereas `RESOURCE_MY_PROFILE` only applies to the logged in user.

**Review by kaxil — APPROVED** [review]

File: airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py (hunk 2 above)
Comment (jhtimmins) [reply airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py:141]: @ashb What would be the purpose of that?

Comment (ashb) [reply airflow/migrations/versions/a13f7613ad25_resource_based_permissions_for_default_.py:141]: I honestly couldn't tell you anymore.

**Pull request merged**